src/lib/factory.js

```diff
@@ -201,6 +201,11 @@ class PluginFactory {
       account: account
     })
 
+    // make sure that the plugin uses admin credentials
+    plugin.credentials.username = this.adminUsername
```

Comment:
Overriding this immediately after calling the constructor is haphazard. Maybe add an (optional) `credentials` parameter to allow the factory to override the default credentials assignment.